modules/extension/app-schema/app-schema/src/main/java/org/geotools/data/complex/AppSchemaDataAccess.java

```diff
@@ -347,25 +341,22 @@ private boolean canCount(
             // we build each Feature
             if (postFilter == null || postFilter.equals(Filter.INCLUDE)) {
                 canCount = true;
+            } else {
+                LOGGER.log(Level.FINE, "Skipping count query due Complex post filter");
             }
         }
         return canCount;
     }
 
-    private int getCountJoin(
-            JoiningJDBCFeatureSource source,
-            Query targetQuery,
-            FeatureTypeMapping mapping,
-            Set<PropertyName> propertyNames)
+    private int getCountNestedFilter(
+            JoiningJDBCFeatureSource source, Query targetQuery, FeatureTypeMapping mapping)
             throws IOException {
-        List<FeatureChainedAttributeDescriptor> featureChainDescriptors =
-                getFeatureChainedDescriptorsFromFilter(mapping, propertyNames);
         // wrap the query in the Joining one to have the unmapping visitor
         // setting the ids
         JoiningQuery unrollQuery =
                 (JoiningQuery) unrollQuery(new JoiningQuery(targetQuery), mapping);
-        JoiningQuery joining = buildJoiningQuery(featureChainDescriptors, unrollQuery, mapping);
-        return source.getCount(joining);
+        unrollQuery.setRootMapping(mapping);
```

Comment:
Joining query instance is not used anymore, is this safe for legacy functionality?  Please test these changes with postgis integration tests.

Reply:
Same as above.